strset/strset_test.go

```diff
@@ -0,0 +1,107 @@
+/*
+
+
+Copyright 2020 Red Hat, Inc.
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU Affero General Public License as
+published by the Free Software Foundation, either version 3 of the
+License, or (at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Affero General Public License for more details.
+
+You should have received a copy of the GNU Affero General Public License
+along with this program.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+package strset
+
+import (
+	"testing"
+)
+
+func TestAdd(t *testing.T) {
+	strSet := NewSet()
+	strSet.Add("key1")
+	strSet.Add("key2")
+	strSet.Add("key3")
+	if strSet.Len() != 3 {
+		t.Errorf("expected strSet %v to have length 3 but got %d", strSet, strSet.Len())
+	}
+}
+
+func TestContains(t *testing.T) {
+	strSet := NewSet()
+	strSet.Add("key1")
+	if strSet.Contains("key1") == false {
+		t.Errorf("expected strSet %v to contain key 'key1' but it does not", strSet)
+	}
+	if strSet.Contains("key2") == true {
+		t.Errorf("expected strSet %v to not contain key 'key2' but it does", strSet)
+	}
+}
+
+func TestRemove(t *testing.T) {
+	strSet := NewSet()
+	strSet.Add("key1")
+	strSet.Add("key2")
+	if strSet.Contains("key1") == false {
+		t.Errorf("expected strSet %v to contain 'key1' but it does not", strSet)
+	}
+	if strSet.Contains("key2") == false {
+		t.Errorf("expected strSet %v to contain 'key2' but it does not", strSet)
+	}
+	if strSet.Len() != 2 {
+		t.Errorf("expected strSet %v to have length 2 but got %d", strSet, strSet.Len())
+	}
+
+	strSet.Remove("key0")
```

Comment:
Should we also check that this key does not exist above here?

Reply:
Good point! Definitely yes